File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let max = 0, tm = 0;
    let val = prices[0];
    for (let i = 0; i < prices.length; i++) {
        if (val > prices[i]) val = prices[i];
        tm = prices[i] - val;
        max = Math.max(tm, max);
    }
    return max;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
       int max = 0, temp = 0, val = prices[0];
       for(int i : prices){
        if(val > i){
            val = i; 
        }
        temp = i - val;
        max = Math.max(temp,max);
       }
       return max;
    }
}